### PR TITLE
Fix aiohttp retry test

### PR DESCRIPTION
**Pull request by Stevenjin8**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ x **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ async def _initial_request(self):
                 self.size = self._file_size
 
         except HttpResponseError as error:
-            if self._start_range is None and error.response.status_code == 416:
```
Comment (Stevenjin8) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:354]: `response` is an optional field for `HttpResponseError`. When passed, it sets `self.status_code = self.response.status_code`.

https://github.com/Azure/azure-sdk-for-python/blob/a23a8f6f100b27629ee6f2c326ed66a2938bd908/sdk/core/azure-core/azure/core/exceptions.py#L292-L301

Comment (Stevenjin8) [reply sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:354]: and now we no longer get an attribute error when `error.response = None`

**Review by jalauzon-msft — APPROVED** [review]
Thanks for getting this fixed up!

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py (hunk 1 above)
Comment (jalauzon-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/aio/_download_async.py:354]: Cool, makes sense.

**Pull request merged**